Source/Core/GoogleEarthEnterpriseTerrainData.js

```diff
@@ -0,0 +1,521 @@
+/*global define*/
+define([
+    '../ThirdParty/when',
+    './BoundingSphere',
+    './Cartesian2',
+    './Cartesian3',
+    './defaultValue',
+    './defined',
+    './defineProperties',
+    './DeveloperError',
+    './GeographicTilingScheme',
+    './HeightmapTessellator',
+    './IndexDatatype',
+    './Intersections2D',
+    './Math',
+    './OrientedBoundingBox',
+    './QuantizedMeshTerrainData',
+    './Rectangle',
+    './TaskProcessor',
+    './TerrainEncoding',
+    './TerrainMesh',
+    './TerrainProvider'
+], function(
+    when,
+    BoundingSphere,
+    Cartesian2,
+    Cartesian3,
+    defaultValue,
+    defined,
+    defineProperties,
+    DeveloperError,
+    GeographicTilingScheme,
+    HeightmapTessellator,
+    IndexDatatype,
+    Intersections2D,
+    CesiumMath,
+    OrientedBoundingBox,
+    QuantizedMeshTerrainData,
+    Rectangle,
+    TaskProcessor,
+    TerrainEncoding,
+    TerrainMesh,
+    TerrainProvider) {
+    'use strict';
+
+    /**
+     * Terrain data for a single tile from a Google Earth Enterprise server.
+     *
+     * @alias GoogleEarthEnterpriseTerrainData
+     * @constructor
+     *
+     * @param {Object} options Object with the following properties:
+     * @param {ArrayBuffer} options.buffer The buffer containing terrain data.
+     * @param {Number} [options.childTileMask=15] A bit mask indicating which of this tile's four children exist.
+     *                 If a child's bit is set, geometry will be requested for that tile as well when it
+     *                 is needed.  If the bit is cleared, the child tile is not requested and geometry is
+     *                 instead upsampled from the parent.  The bit values are as follows:
+     *                 <table>
+     *                  <tr><th>Bit Position</th><th>Bit Value</th><th>Child Tile</th></tr>
+     *                  <tr><td>0</td><td>1</td><td>Southwest</td></tr>
+     *                  <tr><td>1</td><td>2</td><td>Southeast</td></tr>
+     *                  <tr><td>2</td><td>4</td><td>Northeast</td></tr>
+     *                  <tr><td>3</td><td>8</td><td>Northwest</td></tr>
+     *                 </table>
+     * @param {Boolean} [options.createdByUpsampling=false] True if this instance was created by upsampling another instance;
+     *                  otherwise, false.
+     *
+     *
+     * @example
+     * var buffer = ...
+     * var childTileMask = ...
+     * var terrainData = new Cesium.GoogleEarthEnterpriseTerrainData({
+     *   buffer : heightBuffer,
+     *   childTileMask : childTileMask
+     * });
+     *
+     * @see TerrainData
+     * @see HeightTerrainData
+     * @see QuantizedMeshTerrainData
+     */
+    function GoogleEarthEnterpriseTerrainData(options) {
+        //>>includeStart('debug', pragmas.debug);
+        if (!defined(options) || !defined(options.buffer)) {
+            throw new DeveloperError('options.buffer is required.');
```

Comment:
Same comment as earlier about the `options` object and using `Check` here, I'll stop commenting on it, just fix it throughout.